### ISSUE TITLE
UriTemplate redundantly percent encodes literal part

### ISSUE THREAD

**Issue by bodewig** [spring-projects/spring-hateoas#1127]
When using `UriTemplate` with a literal part using reserved characters the part is percent encoded twice. For example

```java
new UriTemplate("https://example.org/foo and bar/{baz}").expand("xyzzy")
```

yields `https://example.org/foo%2520and%2520bar/xyzzy` rather than the expected `https://example.org/foo%20and%20bar/xyzzy`. I think this happens because `UriTemplate.expand` first constructs and expands a Spring Framework `UriTemplate` from the literal part (which adds percent encoding), builds a `UriComponentsBuilder` from the result and later calls `build` with no arguments on this `UriComponentsBuilder` which adds another layer of percent encoding. I think it would be possible to percent encode all variable values beforehand and then invoke `build(false)` instead, but I haven't checked all code paths.

Note that when using UriTemplate with a properly percent-encoded literal part to begin with, things become even worse because of https://github.com/spring-projects/spring-framework/issues/24094. 

```java
new UriTemplate("https://example.org/foo%20and%20bar/{baz}").expand("xyzzy")
```

yields `https://example.org/foo%252520and%252520bar/xyzzy`.

**Comment by odrotbohm** [comment]
I followed @rstoyanchev's advice from spring-projects/spring-framework#24094 and avoid Spring's `UriTemplate` entirely. Also backported to 1.0.x via #1130.

**Comment by gregturn** [comment]
Possibly relayed to #593. If so, and if resolved, there’s a great test case we’ll be a me to add there to avoid regressions.

**Comment by odrotbohm** [comment]
I revisited encoding related tickets and am just polishing up a fix for #593.

**Comment by bodewig** [comment]
Great, many thanks @odrotbohm !
